### DISCUSSION TITLE
`inf` in return values

### DISCUSSION THREAD

**Discussion by nsteins** [open-meteo/open-meteo#395] [Ideas]
I have encountered responses from the API with the value of `inf`. Apparently, this is technically invalid JSON, which does not support nan or infinite values. As a result, I am not able to decode the response using Python's `json.loads`. Python does  allow the parsing of infinite values, but only if they are labeled `Infinity`. I was able to work around this by using string replacement for now, but this may be an issue for others going forward. You can see an example of this issue on another project here: https://github.com/graphite-project/graphite-web/issues/813

**Comment by patrick-zippenfenig** [comment]
The API is not supposed to return infinite values. Could you send the API query for which you got them? Thanks

**Reply by nsteins** [reply]
`{'latitude': 37.69186299999999, 'longitude': -105.99080000000002, 'hourly': 'cloudcover,cloudcover_low,cloudcover_mid,cloudcover_high,direct_radiation,diffuse_radiation,direct_normal_irradiance,shortwave_radiation,temperature_2m,pressure_msl,windspeed_80m,windspeed_120m,winddirection_80m,winddirection_120m', 'start_date': '2022-11-16', 'end_date': '2023-05-28'}`

**Reply by patrick-zippenfenig** [reply]
I still do not see any `inf`. I tried [this URL](https://api.open-meteo.com/v1/forecast?latitude=37.691863&longitude=-105.99&hourly=cloudcover,cloudcover_low,cloudcover_mid,cloudcover_high,direct_radiation,diffuse_radiation,direct_normal_irradiance,shortwave_radiation,temperature_2m,pressure_msl,windspeed_80m,windspeed_120m,winddirection_80m,winddirection_120m&start_date=2022-11-16&end_date=2023-05-01). Could you send the API URL and pin point the issue a bit futher down? Thanks!

**Reply by nsteins** [reply]
Here is the full URL, apologies:

https://api.open-meteo.com/v1/forecast?latitude=37.69186299999999&longitude=-105.99080000000002&hourly=cloudcover%2Ccloudcover_low%2Ccloudcover_mid%2Ccloudcover_high%2Cdirect_radiation%2Cdiffuse_radiation%2Cdirect_normal_irradiance%2Cshortwave_radiation%2Ctemperature_2m%2Cpressure_msl%2Cwindspeed_80m%2Cwindspeed_120m%2Cwinddirection_80m%2Cwinddirection_120m&start_date=2022-11-16&end_date=2023-05-28

**Reply by patrick-zippenfenig** [reply]
The issue is now fixed. 2 Causes:
1. the DNI calculation had a division by 0 on very rare occurrences https://github.com/open-meteo/open-meteo/commit/9e093c702390a7bc9217d9b71e4858c3b9af385f. This was introduced 3 weeks ago while optimising the solar position calculations to make DNI more accurate
2. The JSON formatter code allowed "inf" and "-inf" instead of converting them to NaN. I am now checking if a value if a value is finate instead of just checking for NaN values https://github.com/open-meteo/open-meteo/commit/cbc9cbb5b4aa01a3d8115a53570f4471c44c4883
